spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -1294,6 +1307,16 @@ public void run() {
 			boolean failedAuthRetry = false;
 			this.lastReceive = System.currentTimeMillis();
 			while (isRunning()) {
+
+				try {
+					handleAsyncFailure();
+				}
+				catch (Exception e) {
+					// TODO: Need to improve error handling.
+					// TODO: Need to determine how to handle a failed message.
+					this.logger.error("Failed to process re-try messages. ");
```

Comment:
The logging is enough. 
Exactly same as we do for sync retry logic.
That's what I meant in the my comment from yesterday where you got doubts.